### PR TITLE
Task/use backend morecast token

### LINKED ISSUE #3802: Morecast: Migrate to using API token for posting forecasts to WF1
**Describe the task**
Use our API token for posting forecasts to WF1

**Acceptance Criteria**
- [x] Leverage existing token to use with post forecast calls in the backend 
- [x] Remove frontend workaround for posting forecasts
- [x] Post user IDIR as part of the WF1 forecast

**Additional context**
- Add any other context about the task here.
- Or here

### PR DESCRIPTION
**Pull request by conbrad**
Use our API token with forecast write permissions for morecast forecast posts. Removes frontend WF1 auth wrapper.

Closes #3802 
# Test Links:
[Landing Page](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3824-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/morecast_v2/forecasts.py [hunk 1]
```diff
@@ -75,36 +80,33 @@ async def construct_wf1_forecasts(session: ClientSession, forecast_records: List
     for forecast in forecast_records:
         forecast_timestamp = datetime.fromtimestamp(forecast.for_date / 1000, timezone.utc)
         # Check if an existing daily was retrieved from WF1 and use id and createdBy attributes if present
-        observed_daily = next(
-            (daily for daily in grouped_dailies[forecast.station_code] if daily.utcTimestamp == forecast_timestamp), None)
+        observed_daily = next((daily for daily in grouped_dailies[forecast.station_code] if daily.utcTimestamp == forecast_timestamp), None)
         forecast_id = observed_daily.forecast_id if observed_daily is not None else None
-        created_by = observed_daily.created_by if observed_daily is not None else None
+        created_by = username
```
Comment (dgboss) [inline api/app/morecast_v2/forecasts.py:85]: We're potentially patching an existing forecast. I wonder if we should retain the original `created_by` value and set the `updated_by` field?

Comment (conbrad) [reply api/app/morecast_v2/forecasts.py]: Thanks! Updated in https://github.com/bcgov/wps/pull/3824/commits/c17da1cf4820f6b6c9e4168220afb6e81a2b9313

**Review by dgboss — CHANGES_REQUESTED** [review]
I think a few more changes are required on the front-end. I believe `wf1AuthenticationSlice.ts` can go along with its tests. We can also remove references to the selector from `TabbedDataGrid.tsx` and `rootReducer.ts`.

File: api/app/morecast_v2/forecasts.py (hunk 1 above)
Comment (dgboss) [reply api/app/morecast_v2/forecasts.py]: Do you think we should try setting `lastModifiedBy` to `username`?

**Review by dgboss — APPROVED** [review]
Looks great!

File: api/app/morecast_v2/forecasts.py (hunk 1 above)
Comment (conbrad) [reply api/app/morecast_v2/forecasts.py]: Unfortunately it and `createdBy` get overwritten with our token details.

**Review by brettedw — APPROVED** [review]
Nice work, thanks for tackling that so quickly

**Pull request merged**